File: 1. Easy/118. Pascal's Triangle/Solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();

        if (numRows == 0)
            return triangle;

        triangle.add(Arrays.asList(new Integer[] { 1 }));

        for (int last = 1; last < numRows; last++) {
            List<Integer> prev = triangle.get(last - 1);
            List<Integer> row = new ArrayList<>(prev.size() + 1);

            row.add(1);
            for (int i = 1; i < prev.size(); i++)
                row.add(prev.get(i - 1) + prev.get(i));
            row.add(1);

            triangle.add(row);
        }

        return triangle;
    }
}

File: 1. Easy/118. Pascal's Triangle/solution.js
var generate = function (numRows) {
    var res = [[1]];

    for (let last = 1; last < numRows; last++) {
        let prev = res[last - 1];
        let row = [1];
        for (let i = 1; i < prev.length; i++) {
            row.push(prev[i - 1] + prev[i]);
        }
        row.push(1);
        res.push(row);
    }

    return res;
};